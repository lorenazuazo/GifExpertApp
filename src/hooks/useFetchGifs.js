
import { useEffect, useState } from "react"
import { getGif } from "../helpers/getGif";


export const useFetchGifs = (categoria) => {
    
   const [state, setState] = useState({
       data:[],
       loading:true
   });
    useEffect(()=>{
        getGif(categoria)
            .then(img =>(
                // el tiemaut no va solo fue una prueba
                setTimeout(()=>(
                    setState({
                        data:img,
                        loading:false
                    })
                ),3000)
                
            ))
    },[categoria]);

return state;
}


