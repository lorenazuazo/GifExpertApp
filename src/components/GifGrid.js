import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';
import 'animate.css/animate.min.css';
import PropTypes from 'prop-types';


const GifGrid = ({categoria}) => {

    const {data:images,loading}= useFetchGifs(categoria);

  
    return (
        
    <>
        <h3 className="animate__animated animate__fadeInRight">{categoria}</h3>
        {loading && <p className="animate__animated animate__flash">Cargando...</p>}
        <div className="card-grid">
                {
                    images.map((img) =>(
                    <GifGridItem 
                    key={img.id} 
                    {...img}/>
                    ))
                } 
        </div>
    </>
    )
}

GifGrid.propTypes={
    categoria: PropTypes.string.isRequired
}

export default GifGrid;
