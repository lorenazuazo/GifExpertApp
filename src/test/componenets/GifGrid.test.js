import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')//es para simular/falsear la data que trae el componenete


describe('Pruebas a <GifGrid />', () => {

    const categoria = 'Hola mundo';
   
    
    test('el componente se debe mostrar correctamente', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        })
        const wrapper = shallow(<GifGrid categoria={categoria} />); 
        expect(wrapper).toMatchSnapshot();   
    });
    
    test('Debe de mostrar items cuando se cargan imagenes con useFetchGifs', () => {
        const gif = [{
            id :'1',
            url:'https://localhost/algo.jpg',
            title:'Test al componente'
        },
        {
            id :'2',
            url:'https://localhost/algo.jpg',
            title:'cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data:gif,
            loading:false
        });

        const wrapper = shallow(<GifGrid categoria={categoria} />);
        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gif.length);
    });    
    
});
