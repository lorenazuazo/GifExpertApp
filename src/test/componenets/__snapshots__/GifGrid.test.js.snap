// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <GifGrid /> el componente se debe mostrar correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInRight"
  >
    Hola mundo
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
